apps/costcontrol/js/message_handler.js

```diff
@@ -151,116 +175,68 @@
-        goToTopUpCode = function _goToTopUpCode() {
-          if (inApplicationMode()) {
-            app.launch();
-            window.parent.BalanceTab.topUpWithCode(true);
-          } else {
-            activity = new MozActivity({ name: 'costcontrol/balance' });
-          }
-        };
-      }
+    sendNotification(topUpActivity, title, message, notificationData);
 
-      iconURL += '?topUpError';
-      NotificationHelper.send('topup-incorrectcode-title2', {
-        'bodyL10n': 'topup-incorrectcode-message3',
-        'icon': iconURL
-      }).then(function(notification) {
-        notification.addEventListener('click', goToTopUpCode);
-      });
-
-      if (callback) {
-        callback();
-      }
-    };
+    if (callback) {
+      callback();
+    }
   }
 
   function sendBalanceThresholdNotification(remaining, settings, callback) {
-    // XXX: Hack hiding the message class in the icon URL
-    // Should use the tag element of the notification once the final spec
-    // lands:
-    // See: https://bugzilla.mozilla.org/show_bug.cgi?id=782211
-    navigator.mozApps.getSelf().onsuccess = function _onAppReady(evt) {
-      var app = evt.target.result;
-      var iconURL = NotificationHelper.getIconURI(app);
-
-      var goToBalance;
-      if (!inStandAloneMode()) {
-        goToBalance = function _goToBalance() {
-          if (inApplicationMode()) {
-            app.launch();
-            window.parent.CostControlApp.showBalanceTab();
-          } else {
-            activity = new MozActivity({ name: 'costcontrol/balance' });
-          }
-        };
-      }
 
-      debug('Low limit already notified:', settings.lowLimitNotified);
-      debug('Zero balance already notified:', settings.lowLimitNotified);
+    debug('Low limit already notified:', settings.lowLimitNotified);
+    debug('Zero balance already notified:', settings.lowLimitNotified);
 
-      // Zero reached notification
-      var type;
-      if (remaining.balance === 0 && !settings.zeroBalanceNotified) {
-        type = 'zeroBalance';
+    // Zero reached notification
+    var type;
+    if (remaining.balance === 0 && !settings.zeroBalanceNotified) {
+      type = 'zeroBalance';
 
-      // There is a limit an we are below that limit and we did not notified yet
-      } else if (settings.lowLimit &&
-                 remaining.balance < settings.lowLimitThreshold &&
-                 !settings.lowLimitNotified) {
-        type = 'lowBalance';
+    // There is a limit an we are below that limit and we did not notified yet
+    } else if (settings.lowLimit &&
+               remaining.balance < settings.lowLimitThreshold &&
+               !settings.lowLimitNotified) {
+      type = 'lowBalance';
 
-      // No need for notification
-      } else {
-        if (typeof callback === 'function') {
-          setTimeout(callback);
-        }
-        return;
+    // No need for notification
+    } else {
+      if (typeof callback === 'function') {
```

Comment:
Let's try to modify this to be:
```js
} else if (callback) {
  callback();
}
```
We use this in passing `closeIfProceeds`, so may be we need to make `closeIfProceeds` launch a timeout. Not blocking here but a really-nice-to-have.